docs/rfcs/active/dashboards/1778-dashboards.md

```diff
@@ -0,0 +1,117 @@
+It's important to make charts as a top level entity because charts could be shared between different dashboards.
+We'll need to build `Contains` relationships between Dashboard and Chart entities.
+
+### URN Representation
+We'll define two [URNs](../../../what/urn.md): `DashboardUrn` and `ChartUrn`.
+These URNs should allow for unique identification for dashboards and charts even there are multiple dashboarding tools
+are used within a company. Most of the time, dashboards & charts are given unique ids by the used dashboarding tool.
+An example Dashboard URN for Looker will look like below:
+```
+urn:li:dashboard:(Looker,<<dashboard_id>>)
+```
+An example Chart URN for Redash will look like below:
+```
+urn:li:chart:(Redash,<<chart_id>>)
+```
+
+### Chart metadata
+Dashboarding tools generally have different jargon to denote a chart.
+They are called as [Look](https://docs.looker.com/exploring-data/saving-and-editing-looks) in Looker 
+and [Visualization](https://redash.io/help/user-guide/visualizations/visualization-types) in Redash.
+But, irrespective of the name, charts are the different tiles which exists in a dashboard.
+Charts are mainly used for delivering some information in a visual way to make it easily understandable.
+They might be using single or multiple data sources and generally have an associated query running against
+the underlying data source to generate the data that it will present.
+
+Below is a list of metadata which can be associated with a chart:
+
+- Title
+- Description
+- Type (Bar chart, Pie chart, Scatter plot etc.)
+- Input sources
+- Query (and its type)
+- Access level (public, private etc.)
+- Ownership
+- Status (removed or not)
+- Audit info (last modified, last refreshed)
+
+### Dashboard metadata
+Aside from containing a set of charts, dashboards carry metadata attached to them.
+Below is a list of metadata which can be associated with a dashboard:
+
+- Title
+- Description
+- List of charts
+- Access level (public, private etc.)
+- Ownership
+- Status (removed or not)
+- Audit info (last modified, last refreshed)
+
+### Metadata graph
+
+![dashboards_graph](imgs/dashboards_graph.png)
+
+An example metadata graph showing complete data lineage picture is shown above.
+In this picture, `Dash_A` and `Dash_B` are dashboards and they are connected to charts through `Contains` edges.
+`C1`, `C2`, `C3` and `C4` are charts and they are connected to underlying datasets through `DownstreamOf` edges.
+`D1`, `D2` and `D3` are datasets.
+
+## How we teach this
+
+We should create/update user guides to educate users for:
+ - Search & discovery experience (how to find a dashboard in DataHub)
+ - Lineage experience (how to find upstream datasets of a dashboard and how to find dashboards generated from a dataset)
+
+## Alternatives
+
+In this design, there is an underlying assumption that input sources for charts will be a list of datasets.
```

Comment:
Can we at least make the upstream URN as a union so we can add metrics support later if we want?

Reply:
Would it make more sense to just have different lineage aspects in that case? On for datasets, and one for metrics, etc? Then you don't need a union?

Reply:
Possible, but you'll need to maintain multiple aspects & builders that look 90% identical. Not a fun place to be as the number of URN types grow.

Reply:
While not fun, it is more extensible, and helps prevents merge conflicts if these are your internal only aspects.

Reply:
Like, I wonder if our RFCs shouldn't have an `extensibility` section? :)

Reply:
I guess we can change the type of inputs for a chart to a generic Urn [here](https://github.com/linkedin/datahub/pull/1775/files#diff-9e4e89860d537ee7aaf9cc4438291778R31). WDYT?

Reply:
But yeah, we can define a `ChartInput` enum with DatasetUrn (and MetricUrn later on) as well. That would be stricter and better, I guess.

Also, I'm not quite on the side of having duplicate aspects for this.

Reply:
Refer to https://github.com/linkedin/datahub/pull/1775/files#diff-55f19bf92bba781d0244d6487dba29bd